modules/platform/nuxeo-automation/nuxeo-automation-features/src/main/java/org/nuxeo/ecm/automation/core/operations/services/bulk/BulkRunAction.java

```diff
@@ -162,6 +162,14 @@ public BulkStatus run() throws IOException, OperationException {
         }
         if (queryLimit > 0) {
             builder.queryLimit(queryLimit);
+        } else if (action == "automation-ui" && params.containsKey(OPERATION_ID)) {
+            // automation actions from UI can be limited
+            String operationId = (String) params.get(OPERATION_ID);
+            int limit = AutomationBulkActionUi.getQueryLimit(operationId);
+            if (limit > 0) {
+                log.debug("Set queryLimit to {} for automation-ui operationId: {}", limit, operationId);
```

Comment:
`log.debug("Set queryLimit to: {} for automation-ui operationId: {}", limit, operationId);`